axelor-human-resource/src/main/java/com/axelor/apps/hr/service/expense/ExpenseLineCreateServiceImpl.java

```diff
@@ -184,19 +190,29 @@ protected void setKilometricExpenseLineInfo(
   }
 
   protected ExpenseLine createBasicExpenseLine(
-      Project project, Employee employee, LocalDate expenseDate, String comments)
+      Project project, Employee employee, LocalDate expenseDate, String comments, Currency currency)
       throws AxelorException {
     ExpenseLine expenseLine = new ExpenseLine();
     if (expenseDate.isAfter(appBaseService.getTodayDate(null))) {
       throw new AxelorException(
           TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
           HumanResourceExceptionMessage.EXPENSE_LINE_DATE_ERROR);
     }
+
+    setCurrency(currency, expenseLine);
     expenseLine.setProject(project);
     expenseLine.setEmployee(employee);
     expenseLine.setExpenseDate(expenseDate);
     expenseLine.setComments(comments);
     expenseLine.setTotalAmount(BigDecimal.ZERO);
     return expenseLine;
   }
+
+  protected void setCurrency(Currency currency, ExpenseLine expenseLine) {
+    if (currency != null) {
+      expenseLine.setCurrency(currency);
+    } else {
+      expenseLine.setCurrency(AuthUtils.getUser().getActiveCompany().getCurrency());
```

Comment:
please add more nullcheck for this line: in some cases AuthUtils.getUser() can return null, also the user might not have an active company.